src/esphome/light/light_state.cpp

```diff
@@ -197,7 +202,7 @@ void LightState::current_values_as_cwww(float color_temperature_cw, float color_
   *warm_white = gamma_correct(*warm_white, this->gamma_correct_);
 }
 void LightState::loop() {
-  if (this->active_effect_ != nullptr) {
+  if (this->active_effect_ != nullptr && this->transformer_ == nullptr) {
```

Comment:
Please revert this, effects should still be applied when a transformer is active (effects can call transformers)

Reply:
Then it's impossible to flash while an effect is currently running.

Reply:
It is, see random light effect as an example. Light effects do not always write to the state. As seen in the random light effect, they can use the `.turn_on(); ... .perform();` api to call transitions etc.

Reply:
Let me give you an example of what I mean. I have an rgb strip, that I can apply an effect to (let's say rainbow). I also have a user defined service like this:

```
 20   services:
 21     - service: flash
 22       then:
 23         - light.turn_on:
 24             id: light
 25             brightness: 100%
 26             red: 100%
 27             green: 100%
 28             blue: 100%
 29             flash_length: 100ms
```

While I'm running the effect. If I call that service asynchronously, the strip won't flash, it will briefly change brightness, but that's all.

Reply:
I see, but I'd say that's the intended behavior. Flashes (like transitions) should only temporarily affect the light strip, after that the previous state should be exactly continued. So for example flashing red during a rainbow effect should: 
1. Turn strip red
2. Wait for flash time
3. Go back to rainbow effect

If the user wants to cancel the effect they must a) turn off the light or b) call with `effect: "None"`.

Removing `is_continuous` would solve that I think